pkg/identity/github/issuer.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package github
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/sigstore/fulcio/pkg/identity"
+)
+
+type actionsIssuer struct {
+	*oidc.IDTokenVerifier
+}
+
+func NewActionsIssuer(clientID string) (identity.Issuer, error) {
```

Comment:
To confirm, is the plan to keep these in a cache, like the current implementation?

Reply:
Yup absolutely. So should look like something like this:

```Go
// Loading from hard-coded value here, but this would actually
// be loading using the values in the config file eventually
actionsIssuer, err := github.NewActionsIssuer(`sigstore`)
if err != nil {
  // handle error
}

// Actions OIDC client is "cached" here because the issuer is stateful and part of the issuer pool
pool := identity.Pool{
  actionsIssuer,
  // other issuers
}

// Pool gets dependency injected into the grpc server (again all the OIDC clients state is maintained now as part of the pool)
grpcServer := api.NewGRPCCAServer(ctlogClient, certAuthority, pool)
```